index.d.ts

```diff
@@ -0,0 +1,56 @@
+declare module testableUtils {
+    namespace log {
+        function fatal(message: any);
+        function info(message: any);
+        function error(message: any);
+        function debug(message: any);
+        function trace(message: any);
+    }
+
+    interface events {
+        addListener(event: string | symbol, listener: (...args: any[]) => void): this;
+        on(event: string | symbol, listener: (...args: any[]) => void): this;
+        once(event: string | symbol, listener: (...args: any[]) => void): this;
+        removeListener(event: string | symbol, listener: (...args: any[]) => void): this;
+        off(event: string | symbol, listener: (...args: any[]) => void): this;
+        removeAllListeners(event?: string | symbol): this;
+        setMaxListeners(n: number): this;
+        getMaxListeners(): number;
+        listeners(event: string | symbol): Function[];
+        rawListeners(event: string | symbol): Function[];
+        emit(event: string | symbol, ...args: any[]): boolean;
+        listenerCount(type: string | symbol): number;
+        // Added in Node 6...
+        prependListener(event: string | symbol, listener: (...args: any[]) => void): this;
+        prependOnceListener(event: string | symbol, listener: (...args: any[]) => void): this;
+        eventNames(): Array<string | symbol>;
+    }
+
+    function execute(func: Function)
+
+    namespace dataTable {
+        function open()
+        namespace open{
+            function get(i: number);
+            function random();
+            function next(options?: object);
+        }
+    }
+
+    function results()
+
+    namespace results {
+        function toResourceName(url: string)
+    }
```

Comment:
Take a look at `results.js` there are quite a few methods in there. The `toResourceName()` method is then tacked onto the function as a property in addition to many other functions. I followed the first answer [here](https://stackoverflow.com/questions/12766528/build-a-function-object-with-properties-in-typescript) to account for that. I think something like this:

```
var results: { 
  (resource?: string, url?: string): Result; 
  get(options: object): Promise<number>;
  get(name: string, key?: string): Promise<number>;
  waitForCondition(options: Condition): Promise<void>;
  waitForValue(options: ValueCondition): Promise<void>;
  incrementAndWaitForValue(name: string, value: number): Promise<void>;
  barrier(name: string, value: number): Promise<void>;
  toResourceName: (url: string) => string; 
  current: Result;
}

namespace ValueCondition {
  value: number;
  name: string;
  key?: string;
}

namespace Condition {
  condition: (val: number) => boolean;
  name: string;
  key?: string;
}

namespace Result {
  function timing(options: ResultOptions): Promise<void>;
  function timing(name: string, key?: string, val?: number, units?: string): Promise<void>;
  function counter(options: ResultOptions): Promise<void>;
  function counter(name: string, key?: string, val?: number, units?: string): Promise<void>;
  function histogram(options: ResultOptions): Promise<void>;
  function histogram(name: string, key?: string, val?: number, units?: string): Promise<void>;
  function metered(options: ResultOptions): Promise<void>;
  function metered(name: string, key?: string, val?: number, units?: string): Promise<void>;
  function setTraceStatus(status: string): void;
  function markAsSuccess(): void;
  function markAsFailure(): void;
}

namespace ResultOptions {
  name: string;
  key?: string;
  val?: number;
  units?: string;
}
```

Didn't check this at all but something like that.

Reply:
Hmm, I need to check this in details. Did you used any generator? TBH this is also bit new for me.